src/Streamly/Internal/Data/Stream/StreamD/Eliminate.hs

```diff
@@ -167,10 +181,12 @@ parselMx' pstep initial extract (Stream step state) = do
                         let (src0, buf1) = splitAt n (x:getList buf)
                             src  = Prelude.reverse src0
                         gobuf SPEC s (List buf1) (List src) pst1
-                    PR.Done _ b -> return b
+                    PR.Done _ b -> return (b, Stream step_rest (Left (s, buf)))
```

Comment:
We can use ``D.fromList buf `D.cons` Stream step s`` here, it should simplify to the same thing.

Reply:
Need to move append from Nesting to Types module, so haven't done this for now.

Reply:
Can you raise an issue for it then?